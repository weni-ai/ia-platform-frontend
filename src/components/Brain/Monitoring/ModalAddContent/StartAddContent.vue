<template>
  <section class="add-content">
    <UnnnicIcon
      class="add-content__icon"
      :icon="icon"
      :filled="iconFilled"
      size="xl"
      scheme="neutral-soft"
      data-testid="icon"
    />
    <section class="add-content__infos">
      <UnnnicIntelligenceText
        color="neutral-cloudy"
        family="secondary"
        size="body-gt"
        tag="p"
        data-testid="description"
      >
        {{ description }}
      </UnnnicIntelligenceText>
      <UnnnicIntelligenceText
        class="text-sub-description"
        color="neutral-clean"
        family="secondary"
        size="body-md"
        tag="p"
        data-testid="sub-description"
      >
        {{ subDescription }}
      </UnnnicIntelligenceText>
    </section>

    <UnnnicButton
      class="add-content__trigger"
      type="secondary"
      iconLeft="add"
      data-testid="add-button"
      @click="$emit('add')"
    >
      {{ textAddContent }}
    </UnnnicButton>
  </section>
</template>

<script setup>
const props = defineProps({
  description: {
    type: String,
    default: '',
  },
  subDescription: {
    type: String,
    default: '',
  },
  icon: {
    type: String,
    default: 'text_snippet',
  },
  iconFilled: {
    type: Boolean,
    default: true,
  },
  textAddContent: {
    type: String,
    default: '',
  },
});

const emit = defineEmits(['add']);
</script>

<style scoped lang="scss">
.add-content {
  display: grid;
  justify-items: center;
  align-content: center;
  &__icon {
    margin-bottom: $unnnic-spacing-xs;
  }

  &__infos {
    display: grid;
    justify-items: center;
  }

  &__trigger {
    margin-top: $unnnic-spacing-sm;
  }
}
</style>
