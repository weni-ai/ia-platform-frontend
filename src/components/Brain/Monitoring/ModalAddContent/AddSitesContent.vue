<template>
  <section
    v-if="sites.length"
    class="add-sites-content__sites"
  >
    <ul class="sites__list">
      <li
        v-for="(site, index) of sites"
        :key="index"
      >
        <UnnnicFormElement
          :label="$t('content_bases.sites.modal.fields.link.label')"
        >
          <UnnnicInput
            :ref="(el) => (siteRefs[index] = el)"
            v-model="sites[index]"
            :type="!site || validURL(site) ? 'normal' : 'error'"
            :placeholder="
              $t('content_bases.sites.sidebar_add.fields.link.placeholder')
            "
            @input="onSiteInput($event, index)"
          />
        </UnnnicFormElement>
      </li>
    </ul>
    <UnnnicButton
      type="secondary"
      iconLeft="add"
      :disabled="!validURL(sites.at(-1))"
      @click="addEmptySite"
    >
      {{ $t('router.monitoring.modal_add_content.add_another_site') }}
    </UnnnicButton>
  </section>

  <StartAddContent
    v-else
    icon="globe"
    :description="$t('content_bases.sites.title')"
    :subDescription="$t('content_bases.sites.description')"
    :textAddContent="$t('content_bases.sites.add_site')"
    @add="addEmptySite"
  />
</template>

<script setup>
import { computed, ref, watch } from 'vue';
import { validURL } from '@/utils/sites';

import StartAddContent from './StartAddContent.vue';

const emit = defineEmits(['update:model-value']);

const sites = ref([]);
const siteRefs = ref([]);

const onSiteInput = (event, index) => {
  const siteWithoutSpaces = event.target.value.trim().split(/\s+/).join('');
  sites.value[index] = siteWithoutSpaces;
};

function addEmptySite() {
  sites.value.push('');
}

const validSites = computed(() => sites.value.filter((site) => validURL(site)));

watch(validSites, (newSites) => {
  emit('update:model-value', newSites);
});
</script>

<style scoped lang="scss">
.add-sites-content {
  &__sites {
    height: 100%;

    display: grid;
    grid-template-rows: auto 1fr;
    gap: $unnnic-spacing-xs;

    .sites__list {
      margin: 0;
      padding: 0;
      list-style: none;

      display: grid;
      gap: $unnnic-spacing-xs;
    }
  }
}
</style>
