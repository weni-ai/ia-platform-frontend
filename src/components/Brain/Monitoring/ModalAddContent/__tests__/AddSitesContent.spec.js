import { mount } from '@vue/test-utils';
import { describe, it, expect, beforeEach } from 'vitest';

import AddSitesContent from '../AddSitesContent.vue';
import StartAddContent from '../StartAddContent.vue';

describe('AddSitesContent.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = mount(AddSitesContent);
  });

  it('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders StartAddContent when sites array is empty', () => {
    const startContent = wrapper.findComponent(StartAddContent);
    expect(startContent.exists()).toBe(true);
    expect(startContent.props()).toEqual({
      icon: 'globe',
      iconFilled: true,
      description: expect.any(String),
      subDescription: expect.any(String),
      textAddContent: expect.any(String),
    });
  });

  it('renders the inputs of sites when sites exist', async () => {
    wrapper = mount(AddSitesContent, {
      global: { stubs: ['UnnnicInput'] },
    });

    wrapper.vm.addEmptySite();

    await wrapper.vm.$nextTick();

    const siteInputs = wrapper.findAll('[data-testid="site-input"]');
    expect(siteInputs.length).toBe(1);
  });

  it('adds a new site when the add button is clicked', async () => {
    wrapper.vm.addEmptySite();

    await wrapper.vm.$nextTick();

    const addButton = wrapper.find('[data-testid="add-button"]');
    expect(addButton.exists()).toBe(true);

    const input = wrapper.find('input');
    await input.setValue('http://example.com');

    await addButton.trigger('click');

    const siteList = wrapper.findAll('[data-testid="site-item"]');
    expect(siteList.length).toBe(2);
    expect(siteList[1].find('input').element.value).toBe('');
  });

  it('emits the correct valid sites when a valid site is input', async () => {
    wrapper.vm.addEmptySite();

    await wrapper.vm.$nextTick();

    const input = wrapper.find('input');
    await input.setValue('http://example.com');

    const emitted = wrapper.emitted('update:model-value');
    expect(emitted).toBeTruthy();
    expect(emitted[1]).toEqual([['http://example.com']]);
  });
});
