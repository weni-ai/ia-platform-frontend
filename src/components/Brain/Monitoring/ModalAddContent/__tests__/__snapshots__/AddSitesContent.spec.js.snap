// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddSitesContent.vue > matches the snapshot 1`] = `
"<section data-v-c26a5569="" data-v-c46ee857="" class="add-content" data-testid="start-add-content"><span data-v-76ceda5d="" data-v-c26a5569="" class="material-symbols-rounded unnnic-icon-scheme--neutral-soft unnnic-icon-size--xl material-symbols-rounded--filled add-content__icon" data-testid="icon" translate="no">globe</span>
  <section data-v-c26a5569="" class="add-content__infos">
    <p data-v-ec8da4ad="" data-v-c26a5569="" class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy" data-testid="description">Enter URLs of websites from which the agent can extract information</p>
    <p data-v-ec8da4ad="" data-v-c26a5569="" class="unnnic-text unnnic-text--size-body-md unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-clean text-sub-description text-sub-description" data-testid="sub-description">Make sure the sites are trustworthy and contain relevant data</p>
  </section><button data-v-ed84dbab="" data-v-c26a5569="" class="add-content__trigger unnnic-button unnnic-button--size-large unnnic-button--secondary add-content__trigger" data-testid="add-button">
    <!----><span data-v-76ceda5d="" data-v-ed84dbab="" class="material-symbols-rounded unnnic-icon-scheme--neutral-dark unnnic-icon-size--md unnnic-button__icon-left" data-testid="icon-left" translate="no">add</span>
    <!----><span data-v-ed84dbab="" class="unnnic-button__label" data-testid="button-label">Add website </span>
    <!---->
  </button>
</section>"
`;
